src/sliceDataStorage.h

```diff
@@ -144,9 +156,29 @@ class SliceLayer
 };
 
 /******************/
+
+/*!
+ * A SupportInfillPart represents an connected area for support infill on a layer.
+ * The support infill areas on a layer can be isolated islands, and in this case, a SupportInfillPart represents a single island.
+ *
+ * This data structure is required for gradual support, which needs to partition a support area into a number of sub-areas with different density.
+ * Because support is handled as a whole in the engine, that is, we have a global support areas instead of support areas for each mesh.
+ * With this data structure, we can keep track of which gradual support infill areas belongs to which support area, so we can print them together.
+ */
+class SupportInfillPart
+{
+public:
+    Polygons outline;  //!< outline of this support infill area
+    std::vector<Polygons> insets;  //!< The insets are also known as perimeters or the walls.
+    Polygons inner_infill_areas; //!< The areas for generating infill pattern
+    std::vector<std::vector<Polygons>> gradual_infill_areas_per_combine_per_density;  //!< a list of separated sub-areas which requires different infill densities and combined thicknesses
```

Comment:
You renamed this to `gradual_infill_areas_per_combine_per_density` rather than `infill_areas_per_combine_per_density`. Actually the 'density' part already catches the same aspect as 'gradual', so now the naming contains reference to the gradual infill twice. I think it should be renamed to `infill_areas_per_combine_per_density`.

The documentation should be really clear in which order these vectors are embedded. Please put more comments here.
If I assume the naming above is correct than I understand this function as that `infill_areas_per_combine_per_density[0]` will give me a vector of areas to be assigned different layer thicknesses.
Is that correct?